### DISCUSSION TITLE
Tips for debugging forge issues

### DISCUSSION THREAD

**Discussion by muihlinn** [magit/forge#648] [Support] [answered]
## Background

At some point I had forge working well with github and all kind of repos. 
After several months hiatus , with the corresponding emacs and magit-related updates, the tokens I had were erased, made a new one, changed it in `.authinfo.gpg`,  and I haven't been able to make it work anymore. 

## What I've done

- I've Read all the manuals and rechecked all configs
- Confirmed that the token has the right permissions as manual states
- Confirmed that I can use the personal token from the command line
- Confirmed that I can use my ssh key with github from magit 
- tried several searchs for similar issues

## What I keep getting

```
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
---
I'm running out of ideas on what is going on , what to try or how to debug this issue to make it work again.

**Comment by tarsius** [comment, marked as answer]
```
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

This error isn't coming from Forge but Git. That makes this more of a Magit issue than a Forge issue, but really it is either a Emacs configuration issue or even a Git configuration issue.

Where are you getting that and is it prefixed with the `git` command that produced that error?

> Confirmed that I can use my ssh key with github from magit

So you can `git fetch` on the command line, but the equivalent `magit-fetch` *also* works? The above error indicates that `magit-fetch` does not work. (That's why I asked above about how you got that error.)

It has been a long time since someone last asked me for help with this, so I don't exactly remember where all the footguns are. Generally speaking if it works in a shell but not Emacs, the issue is that the user configured the shell but did not configure Emacs accordingly.

A first thing you can try is to type `emacs &` in a shell, instead of starting it by clicking on an icon or something like that. This may work because it causes Emacs to inherit the environment of the shell.

A common solution to the problem of Emacs not just magically picking up the configuration of the shell, seem to be the `exec-path-from-shell` package. But that too needs a bit of additional setup, the settings mentioned at the end of https://github.com/purcell/exec-path-from-shell?tab=readme-ov-file#usage.

**Reply by muihlinn** [reply]
Thre is something weird here about it, but everything seems correct.

  I noticed that during the hiatus I changed the default ubuntu for i3, so it makes sense that something in the env may have changed, but I haven't changed either shell or emacs config about magit or env in a very long while. Anyway I tried, and fails equally.

I get the same error either fetching `f e remote` or trying to do it with N commands and trying to fetch it from gh. Other LAN remotes work just fine.

Ironically if I clone the repo from the command line and then manage it from magit I can use magit to fetch `origin` but N commands fails equally.  What remote to use, when it's needed, is specified locally in git config. Not sure what's going on there or how to set a trap to find out what causes it, but doesn't smell like a bug to me, more like an obscure issue with things I cannot figure.

**Comment by muihlinn** [comment]
main culprit wasn't shell, was something involving ssh identities and the upstream url which - yet - I cannot explain why didn't happened before. It's working now but I'm not sure if I found the underlying issue or just a way to circumvent it. Thanks much!

**Reply by tarsius** [reply]
I feel for you. I too have to soon investigate an hairy ssh issue I've been suffering from for a while now. :frowning_face:
